src/kbmod/batch_search.py

```diff
@@ -0,0 +1,40 @@
+class BatchSearchManager:
```

Comment:
The main idea here was that I didn't want the user to have to call `prepare_batch_search` and `finish_search`

Reply:
Not for this PR, but something to consider as a future extension: In theory this could be extended further by making this a generator:

```
for batch_search in BatchSearchManager(stack_search, search_list, min_observations, x_size, y_size):
    batch_results.extend(batch_search.search_batch())
```

where the `BatchSearchManager` tracks the block sizes to search, current block locations, etc.